src/App.tsx

```diff
@@ -15,36 +17,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  name = 'name',
+  length = 'length',
+}
+
+interface FillterParams {
+  field: SortType | '';
+  isReverse: boolean;
+}
+
+const prepareGoods = (goods: string[], { field, isReverse }: FillterParams) => {
+  const preparedGoods = [...goods];
+
+  if (field) {
+    preparedGoods.sort((good1, good2) => {
+      switch (field) {
+        case SortType.name:
+          return good1.localeCompare(good2);
+        case SortType.length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return isReverse ? preparedGoods.reverse() : preparedGoods;
+};
+
 export const App: React.FC = () => {
+  const [reverse, setReverse] = useState(false);
+  const [sortField, setSortField] = useState<SortType | ''>('');
+  const showGoods = prepareGoods(goodsFromServer, {
```

Comment:
```suggestion
  const visisbleGoods = prepareGoods(goodsFromServer, {
```
`showGoods` sounds like a function that show goods